personalized_nlp/experiments/goemo_userid_notune.py

```diff
@@ -0,0 +1,132 @@
+    datamodule_cls = GoEmotionsDataModule
+    embedding_types = ['roberta']
+    model_types = ['userid']
+
+    fold_nums = 10
+
+    min_word_counts = [50]
+    words_per_texts = [15]
+
+    batch_size = 16
+    dp_embs = [0.25]
+    embedding_dims = [50]
+    lr_rates = [1e-3]
+    weight_decay = 0.01
+    # set nr_frozen_epochs = 0 to finetuning from scratch, = epochs to frozen the whole
+    epochs = 20
+    nr_frozen_epochs = 20
+
+    user_folding = True
+    use_cuda = True
+    # frozen=False by default for finetuning, set frozen=True for not finetuning
+    frozen = True
+
+    for (min_word_count, words_per_text, embeddings_type, lr_rate) in product(
+        min_word_counts, words_per_texts, embedding_types, lr_rates
+    ):
+
+        seed_everything()
+        data_module = datamodule_cls(
+            embeddings_type=embeddings_type, normalize=regression, batch_size=batch_size,
+        )
+        data_module.prepare_data()
+        data_module.setup()
+        data_module.compute_word_stats(
+            min_word_count=min_word_count,
+            min_std=0.0,
+            words_per_text=words_per_text,
+        )
+
+        for model_type, embedding_dim, dp_emb, fold_num in product(
+            model_types, embedding_dims, dp_embs, range(fold_nums)
+        ):
+            hparams = {
+                "dataset": type(data_module).__name__,
+                "model_type": model_type,
+                "embeddings_type": embeddings_type,
+                "embedding_size": embedding_dim,
+                "fold_num": fold_num,
+                "regression": regression,
+                "words_per_texts": words_per_text,
+                "min_word_count": min_word_count,
+                "dp_emb": dp_emb,
+                "weight_decay": weight_decay,
+                "nr_frozen_epochs": nr_frozen_epochs,
+                "learning_rate": lr_rate,
+            }
+
+            logger = pl_loggers.WandbLogger(
+                save_dir=LOGS_DIR,
+                config=hparams,
+                project=wandb_project_name,
+                log_model=False,
+            )
+
+            output_dim = len(data_module.class_dims) if regression else sum(data_module.class_dims)
+            text_embedding_dim = data_module.text_embedding_dim
+            model_cls = models_dict[model_type]
+
+            model = model_cls(
+                output_dim=output_dim,
+                text_embedding_dim=text_embedding_dim,
+                word_num=data_module.words_number,
+                annotator_num=data_module.annotators_number,
+                dp=0.0,
+                dp_emb=dp_emb,
+                embedding_dim=embedding_dim,
+                hidden_dim=100,
+                bias_vector_length=len(data_module.class_dims),
+                nr_frozen_epochs=nr_frozen_epochs,
+                embedding_type=embeddings_type,
+                frozen = frozen
+            )
+
+            for name, param in model.model.named_parameters():
```

Comment:
It can be kept here but I presume that the same thing will be executed in `freeze` method of the lightning module since `nr_frozen_epochs` is greater than 0.